File: 03_TimeComplexity_TapeEquilibrium.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(a)
    # write your code in Ruby 2.2
    left = a[0]
    right = a.inject(:+) - a[0]
    min = (left - right).abs
    1.upto(a.length - 2) do |i|
      left += a[i]
      right -= a[i]
      diff = (left - right).abs 
      min = diff if min > diff
    end
    return min
end
  

File: Python/03_TimeComplexity_TapeEquilibrium.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # write your code in Python 3.6
    left = A[0]
    right = sum(A) - A[0]
    ans = abs(left - right)
    for i in range(1, len(A) - 1):
        left += A[i]
        right -= A[i]
        ans = min(ans, abs(left - right))

    return ans 
# end of func